CONTRIBUTING.md

```diff
@@ -0,0 +1,316 @@
+  that starts the argument list of a function call; or the open braket
+  that starts an indexing.  In other words, prefer these:
+
+  ```lua
+  foo(bar, baz)
+  foo{ bar, baz }
+  foo"bar"
+  foo[[bar]]
+  foo[bar]
+  ```
+
+* Trailing at the end of line or (newline) at the end of file.
+
+Always surround these binary operators with a single space on either side:
+assignment (`=`), comparisons, Booleans (`and`, `or`, `not`).
+If operators with different priorities are used, consider adding whitespace
+around the operators with the lowest priorities. Use your own judgment;
+however, never use more than one space, and always have
+the same amount of whitespace on both sides of a binary operator.
+
+### Indentation
+
+Use 4 *spaces* per indentation level.
+
+Continuation lines should align wrapped elements either vertically
+inside parentheses, brackets and braces, or using a hanging indent
+(the opening parenthesis of a parenthesized statement is the last
+non-whitespace character of the line, with subsequent lines being indented
+until the closing parenthesis), e.g.
+
+```lua
+-- Vertically aligned
+long_function_call{ foo, bar,
+                    baz }
+
+-- Hanging indentation
+long_function_call(
+    foo, bar
+    baz)
+```
+
+The closing brace or braket on multiline constructs may either line up under
+the first character of the line that starts the construct, as in:
+
+```lua
+long_function_call{
+    foo = 1, bar = 2,
+    baz = 3,
+}
+```
+
+In this case, and this case only, the trailing comma is acceptable
+to avoid diff noises when more values are added,
+but since Vicious often deal with system APIs which rarely ever change,
+it's occasionally helpful to do so.
+
+Trailing right parentheses, however, are not allowed.
+
+### Maximum Line Length
+
+Limit all *code* lines to a maximum of 80 characters.
+Comments and long strings need not to follow this restriction.
+
+As one might have noticed, the syntatic sugars `f{<fields>}`
+(for `f({<fields>})`) and `f'<string>'` (or `f"<string>"`/`f[[<string>]]`,
+for `f('<string>')`) are espescially prefered to squeeze
+the line length to this limit.
+
+### Blank Lines
+
+Surround function definitions with a single blank line.  Extra blank lines
+may be used (sparingly) to separate groups of related functions.
+Blank lines may be omitted between a bunch of related one-liners
+(e.g. a set of dummy implementations).  
+Use blank lines in functions, sparingly, to indicate logical sections.
+
+### Requiring Libraries
+
+All standard libraries should be localized before used
+for the matter of performance.
+
+`require`s should always be put at the top of the source file,
+just after the copyright header, and before module globals and constants,
+and grouped in the following order:
+
+1. Standard libraries
+2. Related third-party libraries
+3. Local libraries
+
+For example,
+
+```lua
+local type = type
+local table = { concat = table.concat, insert = table.insert }
+
+local awful = require"awful"
```

Comment:
Is the shorthand version ie. foo"bar" instead of foo("bar") preferred?

Reply:
I would not say so: both are OK.  The shorthand is especially preferred when we're running out of columns though.